src/download_fossa_cli.rs

```diff
@@ -0,0 +1,250 @@
+    format!("https://github.com/fossas/fossa-cli/releases/download/v{version}/fossa_{version}_windows_amd64.zip")
+}
+
+#[cfg(target_os = "macos")]
+fn download_url(version: &str) -> String {
+    format!("https://github.com/fossas/fossa-cli/releases/download/v{version}/fossa_{version}_darwin_amd64.zip")
+}
+
+#[cfg(target_os = "linux")]
+fn download_url(version: &str) -> String {
+    format!("https://github.com/fossas/fossa-cli/releases/download/v{version}/fossa_{version}_linux_amd64.zip")
+}
+
+#[tracing::instrument]
+async fn download_from_github(version: &str) -> Result<Cursor<Bytes>, Error> {
+    let client = reqwest::Client::new();
+    let response = client
+        .get(download_url(version))
+        .send()
+        .await
+        .into_report()
+        .change_context(Error::Download)
+        .help_lazy(|| indoc!{"
+            Try downloading FOSSA CLI from '{download_url}' to determine if this is an issue with the local network.
+            You also may be able to work around this issue by using the installation script for FOSSA CLI,
+            located at https://github.com/fossas/fossa-cli#installation
+            "}
+        )?;
+
+    let content = response
+        .bytes()
+        .await
+        .into_report()
+        .change_context(Error::Download)
+        .help_lazy(|| indoc!{"
+            Try downloading FOSSA CLI from '{download_url}' to determine if this is an issue with the local network.
+            You also may be able to work around this issue by using the installation script for FOSSA CLI,
+            located at https://github.com/fossas/fossa-cli#installation
+            "}
+        )?;
+    let content = Cursor::new(content);
+    Ok(content)
+}
+
+#[tracing::instrument(skip(content))]
+async fn unzip_zip(content: Cursor<Bytes>, final_path: &PathBuf) -> Result<(), Error> {
+    let mut archive = zip::ZipArchive::new(content)
+        .context(Error::Extract)
+        .describe("extracting zip file from downloaded fossa release")?;
+    let zip_file = match archive.by_name(command_name()) {
+        Ok(file) => file,
+        Err(..) => {
+            return Err(Error::Extract)
+                .into_report()
+                .describe("finding fossa executable in downloaded fossa release");
+        }
+    };
+
+    write_zip_to_final_file(zip_file, final_path)?;
+    Ok(())
+}
+
+#[tracing::instrument(skip(zip_file))]
+#[cfg(target_family = "windows")]
+fn write_zip_to_final_file<R>(mut zip_file: R, final_path: &PathBuf) -> Result<(), Error>
+where
+    R: Read,
+{
+    let final_path_string = final_path.to_str().unwrap_or("").to_string();
+    let mut final_file = fs::OpenOptions::new()
+        .create(true)
+        .write(true)
+        .open(final_path)
+        .into_report()
+        .change_context(Error::FinalCopy(final_path_string.clone()))?;
+    copy(&mut zip_file, &mut final_file)
+        .into_report()
+        .change_context(Error::FinalCopy(final_path_string))?;
+    Ok(())
```

Comment:
[nit] prefer the `lazy` variant to defer this clone to if an error actually appears.

```suggestion
    let final_path_string = final_path.to_str().unwrap_or("").to_string();
    let mut final_file = fs::OpenOptions::new()
        .create(true)
        .write(true)
        .open(final_path)
        .into_report()
        .change_context_lazy(|| Error::FinalCopy(final_path_string.clone()))?;
    copy(&mut zip_file, &mut final_file)
        .into_report()
        .change_context_lazy(|| Error::FinalCopy(final_path_string))?;
    Ok(())
```